include/open62541/plugin/accesscontrol.h

```diff
@@ -80,27 +108,36 @@ struct UA_AccessControl {
     /* Allow adding a node */
     UA_Boolean (*allowAddNode)(UA_Server *server, UA_AccessControl *ac,
                                const UA_NodeId *sessionId, void *sessionContext,
-                               const UA_AddNodesItem *item);
+                               const UA_AddNodesItem *item, UA_RolePermissionType *userRolePermission, size_t userRoleSize);
```

Comment:
Don't modify the API for allowAddNode, etc.

I see the RolePermissions as an alternative way for controlling the access.
If there are RolePermissions defined for a node (and not all nodes will have them), then we check the RolePermissions internally.

No need to query the AccessControl plugin for this.
Because the server internally has access to the RolePermissions of the node and the roles configured for each session.

If however there are no RolePermissions defined for a node, then we continue to query to AccessControl plugin as before.